examples/finding_sunspots_using_stara.py

```diff
@@ -0,0 +1,130 @@
+
+###############################################################################
+# Finally, we plot the resampled map along with the detected sunspots. We create
+# a new Matplotlib figure and subplot with the projection defined by the resampled
+# map, plot the resampled map, and overlay contours of the detected sunspots using
+# the ``contour`` method.
+
+fig = plt.figure()
+ax = plt.subplot(projection=cont_rotated_resample)
+im = cont_rotated_resample.plot(axes=ax, autoalign=True)
+ax.contour(segs, levels=0)
+
+###############################################################################
+# To focus on specific regions containing sunspots, we can create a submap,
+# which is a smaller section of the original map. This allows us to zoom in
+# on areas of interest. We define the coordinates of the rectangle to crop
+# in pixel coordinates.
+
+bottom_left = cont_rotated_resample.pixel_to_world(240 * u.pix, 350 * u.pix)
+top_right = cont_rotated_resample.pixel_to_world(310 * u.pix, 410 * u.pix)
+
+# Create the submap using the world coordinates of the bottom left and top right corners.
+hmi_submap = cont_rotated_resample.submap(bottom_left, top_right=top_right)
+segs = stara(hmi_submap, limb_filter=10 * u.percent)
+
+# Plot the submap along with the contours.
+fig = plt.figure()
+ax = plt.subplot(projection=hmi_submap)
+im = hmi_submap.plot(axes=ax, autoalign=True)
+ax.contour(segs, levels=0)
+
+plt.show()
+
+###############################################################################
+# We can further enhance our analysis by extracting key properties from the
+# segmented image and organizing them into a structured table.
+
+# First, a labeled image is created where each connected component (sunspot)
+# is assigned a unique label.
+labelled = label(segs)
+# If no sunspots are detected (labelled.max() == 0), print an empty table.
+if labelled.max() == 0:
+    print(QTable())
+else:
+    # Extract properties of the labeled regions (sunspots)
+    regions = regionprops_table(
+        labelled,
+        hmi_submap.data,
+        properties=[
+            "label",  # Unique for each sunspot
+            "centroid",  # Centroid coordinates (center of mass)
+            "area",  # Total area (number of pixels)
+            "min_intensity",
+        ],
+    )
+    # A new column named "obstime" is added to the table, which contains
+    # the observation date for each sunspot.
+    regions["obstime"] = Time([hmi_submap.date] * regions["label"].size)
+    # The pixel coordinates of sunspot centroids are converted to world coordinates
+    # (solar longitude and latitude) in the heliographic Stonyhurst projection.
+    regions["center_coord"] = hmi_submap.pixel_to_world(
+        regions["centroid-0"] * u.pix,
+        regions["centroid-1"] * u.pix,
+    ).heliographic_stonyhurst
+    # Finally, the QTable containing the extracted sunspot properties is printed.
+    print(QTable(regions))
```

Comment:
Can we create a follow up plot that labels the sunspots with the center locations?

Reply:
Sure that might work, let me add that too.